crates/pathfinder/src/rpc/v02/method/get_transaction_by_hash.rs

```diff
@@ -0,0 +1,176 @@
+            let positional = r#"[
+                "0xdeadbeef"
+            ]"#;
+            let positional = Params::new(Some(positional));
+
+            let input = positional.parse::<GetTransactionByHashInput>().unwrap();
+            assert_eq!(
+                input,
+                GetTransactionByHashInput {
+                    transaction_hash: StarknetTransactionHash(starkhash!("deadbeef"))
+                }
+            )
+        }
+
+        #[test]
+        fn named_args() {
+            let named_args = r#"{
+                "transaction_hash": "0xdeadbeef"
+            }"#;
+            let named_args = Params::new(Some(named_args));
+
+            let input = named_args.parse::<GetTransactionByHashInput>().unwrap();
+            assert_eq!(
+                input,
+                GetTransactionByHashInput {
+                    transaction_hash: StarknetTransactionHash(starkhash!("deadbeef"))
+                }
+            )
+        }
+    }
+
+    mod errors {
+        use super::*;
+
+        #[tokio::test]
+        async fn hash_not_found() {
+            let context = std::sync::Arc::new(RpcContext::for_tests());
+            let input = GetTransactionByHashInput {
+                transaction_hash: StarknetTransactionHash(starkhash_bytes!(b"non_existent")),
+            };
+
+            let result = get_transaction_by_hash(context, input).await;
+
+            assert_matches::assert_matches!(
+                result,
+                Err(GetTransactionByHashError::TxnHashNotFound)
+            );
+        }
+    }
+
+    #[tokio::test]
+    async fn success() {
+        let context = std::sync::Arc::new(RpcContext::for_tests());
+        let input = GetTransactionByHashInput {
+            transaction_hash: StarknetTransactionHash(starkhash_bytes!(b"txn 0")),
+        };
+
+        let result = get_transaction_by_hash(context, input).await.unwrap();
+        use crate::rpc::v02::types::reply;
+        assert_eq!(
+            result,
+            Transaction::Invoke(reply::InvokeTransaction::V0(reply::InvokeTransactionV0 {
+                common: reply::CommonInvokeTransactionProperties {
+                    hash: StarknetTransactionHash(starkhash_bytes!(b"txn 0")),
+                    max_fee: Fee(web3::types::H128::zero()),
+                    signature: vec![],
+                    nonce: TransactionNonce(StarkHash::ZERO),
+                },
+                contract_address: ContractAddress::new_or_panic(starkhash_bytes!(b"contract 0")),
+                entry_point_selector: EntryPoint(StarkHash::ZERO),
+                calldata: vec![],
+            }))
+        )
+    }
+
+    #[tokio::test]
+    async fn pending() {
+        let context = RpcContext::for_tests();
+        let pending_data = crate::rpc::tests::create_pending_data(context.storage.clone()).await;
+        let context = context.with_pending_data(pending_data);
+        let context = std::sync::Arc::new(context);
```

Comment:
I wonder if we shouldn't fold this into a single `RpcContext::for_tests_with_pending() -> Arc<RpcContext>`?

Reply:
We can -- I didn't do that because this was the only place this was used. But since other methods will need that for sure I think we could just do that right now.